src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -761,4 +754,199 @@ public static String getSchemaPositionSHA(Long serverID, Position position) {
 		);
 		return DigestUtils.shaHex(shaString);
 	}
+
+	public int getSchemaChainLength() {
+		return schemaChain.length();
+	}
+
+	public int compact(Connection connection, int chainMaxLength, int chainLength) throws SQLException {
```

Comment:
general thoughts on compaction:  

- let's simplify the code by always truncating the schema all the way down.  I don't think there's a use case for keeping chains around.

- If possible (and I'm not sure 100% sure this is right), I think it's better to insert a new base schema with no back-reference than to re-use the old schemas.  This would let us someday insert the new schema synchronously and then lazily delete the old ones.

Reply:
- OK, whole chain is always removed now.
- Inserting new schema would violate unique key on `position_sha`.

Reply:
gotcha, right `position_sha`.  ok.